src/RadixDlt.NetworkGateway.Abstractions/Model/PendingTransactionPayloadLedgerStatus.cs

```diff
@@ -0,0 +1,162 @@
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+using System;
+
+namespace RadixDlt.NetworkGateway.Abstractions.Model;
+
+/// <summary>
+/// This follows the convention from the Core API.
+/// </summary>
+public enum PendingTransactionPayloadLedgerStatus
+{
+    /// <summary>
+    /// We know that this specific payload has been committed as a success.
+    /// </summary>
+    CommittedSuccess,
+
+    /// <summary>
+    /// We know that this specific payload has been committed as a failure.
+    /// </summary>
+    CommittedFailure,
+
+    /// <summary>
+    /// We know that the payload has been committed, but don't yet know the outcome until it's synced (Success/Failure).
+    /// </summary>
+    CommitPendingOutcomeUnknown,
+
+    /// <summary>
+    /// We know that the intent has been committed with a different payload, but don't yet know the outcome
+    /// until it's synced (Success/Failure).
+    /// </summary>
+    CommitOfOtherPayloadForIntentPendingOutcomeUnknown,
+
+    /// <summary>
+    /// We know that this specific payload is permanently rejected.
+    ///
+    /// Note - this does not imply that the intent inside the payload is permanently rejected.
+    /// For example, if there are multiple payloads containing the same intent, only one payload
+    /// gets committed, and the rest get permanently rejected.
+    /// </summary>
+    PermanentlyRejected,
+
+    /// <summary>
+    /// This specific payload was accepted into the mempool at the last execution.
+    /// </summary>
+    TransientlyAccepted,
+
+    /// <summary>
+    /// This specific payload was rejected at the last execution.
+    /// </summary>
+    TransientlyRejected,
+
+    /// <summary>
+    /// We don't know anything about whether the intent could be committed or rejected.
+    /// </summary>
+    Unknown,
+}
+
+public static class PendingTransactionPayloadLedgerStatusExtensions
+{
+    public static bool ShouldStopSubmittingTransactionToNetwork(this PendingTransactionPayloadLedgerStatus status)
+    {
+        return status switch
+        {
+            PendingTransactionPayloadLedgerStatus.CommittedSuccess => true,
+            PendingTransactionPayloadLedgerStatus.CommittedFailure => true,
+            PendingTransactionPayloadLedgerStatus.CommitPendingOutcomeUnknown => true,
+            PendingTransactionPayloadLedgerStatus.CommitOfOtherPayloadForIntentPendingOutcomeUnknown => true,
+            PendingTransactionPayloadLedgerStatus.PermanentlyRejected => true,
+            PendingTransactionPayloadLedgerStatus.TransientlyAccepted => false,
+            PendingTransactionPayloadLedgerStatus.TransientlyRejected => false,
+            PendingTransactionPayloadLedgerStatus.Unknown => false,
+            _ => throw new ArgumentOutOfRangeException(nameof(status), status, null),
+        };
```

Comment:
That's a very Rusty way of coding :D

Reply:
Hah yes! Will change to an abstract method.

Reply:
Actually I can't - because it's an enum. But will change the `editorconfig` to allow us to make enum matching of named variants exhaustive, so a change will be caught as a compiler error.